base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/pythoninterface/PythonInterface.java

```diff
@@ -443,6 +443,37 @@ public void refreshTimeEstimation(List<ScriptGeneratorAction> scriptGenContent,
         }
     }
     
+    /**
+     * Use python to calculate a custom estimation defined by the scriptDefinition for the current parameters and refresh the
+     * custom estimation property.
+     * 
+     * @param scriptGenContent The script generator content
+     * @param scriptDefinition The script definition
+     * @param globalParams The global parameters to refresh custom estimation with.
+     * @throws ExecutionException A failure to execute the py4j call
+     * @throws InterruptedException The Py4J call was interrupted
+     * @throws PythonNotReadyException When python is not ready to accept calls.
+     */
+    public void refreshCustomEstimation(List<ScriptGeneratorAction> scriptGenContent, ScriptDefinitionWrapper scriptDefinition, List<String> globalParams)
```

Comment:
As with many of the other methods in this PR, there's a lot of boilerplate here shared with other methods in this class. Can we refactor tp use common helper methods?